SurveysTests/Sources/Specs/Data/Repositories/AuthenticationRepositorySpec.swift

```diff
@@ -0,0 +1,111 @@
+//
+//  AuthenticationRepositorySpec.swift
+//  SurveysTests
+//
+//  Created by David Bui on 18/11/2022.
+//  Copyright © 2022 Nimble. All rights reserved.
+//
+
+import Combine
+import Nimble
+import Quick
+import Resolver
+
+@testable import Surveys
+
+final class AuthenticationRepositorySpec: QuickSpec {
+
+    override func spec() {
+
+        var networkAPI: NetworkAPIProtocolMock!
+        var repository: AuthenticationRepositoryProtocol!
+
+        describe("An AuthenticationRepository") {
+
+            beforeEach {
+                Resolver.registerAllMockServices()
+                repository = AuthenticationRepository()
+                networkAPI = Resolver.resolve(NetworkAPIProtocol.self) as? NetworkAPIProtocolMock
+            }
+
+            describe("its login() call") {
+
+                let email = "email@email.com"
+                let password = "password"
+                let tokenToTest = APIToken.dummy
+                let errorToTest = TestError.mock
+                let parameter = LoginParameter(
+                    grantType: Constants.GrantType.password.rawValue,
+                    email: email,
+                    password: password,
+                    clientId: Constants.API.clientId,
+                    clientSecret: Constants.API.clientSecret
+                )
+
+                context("when networkAPI.performRequest emits success") {
```

Comment:
it looks a bit wired to me.. 
```suggestion
                context("when networkAPI emits success from performRequest") {
```